cloud-functions/cf-file-scanner/src/main/java/org/broadinstitute/ddp/clamav/Client.java

```diff
@@ -0,0 +1,272 @@
+package org.broadinstitute.ddp.clamav;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.Scanner;
+import java.util.logging.Logger;
+
+public class Client {
+    private static final Logger logger = Logger.getLogger(Client.class.getName());
+
+    private class Defaults {
+        static final int PORT = 3310;
+        static final int SO_TIMEOUT = 30000; // 30 seconds
+        static final int MAX_CHUNK_SIZE = 1000000; // 1MB
+    }
+
+    private enum Delimiter {
+        NULL,
+        NEWLINE;
+
+        public byte[] modeFlag() {
+            switch (this) {
+                case NULL:
+                    return new byte[] { 'z' };
+                case NEWLINE:
+                    return new byte[] { 'n' };
+                default:
+                    return new byte[] {};
+            }
+        }
+
+        public byte[] lineTerminator() {
+            switch (this) {
+                case NULL:
+                    return new byte[] { 0x00 };
+                case NEWLINE:
+                    return new byte[] { 0x0A };
+                default:
+                    return new byte[] {};
+            }
+        }
+    }
+
+    private enum Command {
+        PING("PING"),
+        VERSION("VERSION"),
+        INSTREAM("INSTREAM");
+
+        private final String command;
+
+        private Command(final String command) {
+            this.command = command;
+        }
+
+        public byte[] getBytes() {
+            return this.command.getBytes(StandardCharsets.US_ASCII);
+        }
+    }
+
+    public enum MalwareResult {
+        /** One or more pieces of malware were identified during a scan */
+        POSITIVE,
+
+        /** No malware was identified during a scan*/
+        NEGATIVE;
+    }
+
+    public class ScanResult {
+        public final MalwareResult result;
+        public final Optional<String> message;
+
+        private ScanResult(MalwareResult result) {
+            this.result = result;
+            this.message = Optional.empty();
+        }
+
+        private ScanResult(MalwareResult result, String message) {
+            this.result = result;
+            this.message = Optional.ofNullable(message);
+        }
+    }
+
+    private InetSocketAddress clamdAddress;
+    private int socketTimeout = Defaults.SO_TIMEOUT;
+
+    public Client(String host) {
+        this(host, Defaults.PORT);
+    }
+    
+    public Client(String host, int port) {
+        assert host != null;
+        assert host.isEmpty() == false;
```

Comment:
Replace these with throwing checks, or just leave the verification to `InetSocketAddress`